packages/veraswap-app/src/atoms/hyperlane.ts

```diff
@@ -87,35 +87,35 @@ export const hyperlaneMailboxChainOut = atom((get) => {
 });
 
 export const hypERC20CollateralWrappedTokenQueryAtom = atomWithQuery((get) => {
-    const tokenIn = get(tokenInAtom);
+    const currencyIn = get(currencyInAtom);
 
-    if (!tokenIn) return disabledQueryOptions as any;
+    if (!currencyIn) return disabledQueryOptions as any;
 
-    if (tokenIn.standard !== "HypERC20Collateral") {
+    if (!currencyIn || !isMultichainToken(currencyIn) || currencyIn.isHypERC20() || !currencyIn.hyperlaneAddress) {
         return disabledQueryOptions as any;
     }
 
     return readContractQueryOptions(config, {
-        chainId: tokenIn.chainId,
-        address: tokenIn.address,
+        chainId: currencyIn.chainId,
+        address: currencyIn.address,
         abi: HypERC20Collateral.abi,
         functionName: "wrappedToken",
         args: [],
     });
 }) as Atom<AtomWithQueryResult<Address>>;
 
 export const tokenRouterQuoteGasPaymentQueryAtom = atomWithQuery((get) => {
-    const tokenIn = get(tokenInAtom);
+    const currencyIn = get(currencyInAtom);
     const chainOut = get(chainOutAtom);
 
-    if (!tokenIn || !chainOut) return disabledQueryOptions as any;
-    if (tokenIn.standard !== "HypERC20Collateral") {
+    if (!currencyIn || !chainOut) return disabledQueryOptions as any;
+    if (!isMultichainToken(currencyIn) || currencyIn.isHypERC20() || !currencyIn.hyperlaneAddress) {
```

Comment:
should only check `!isMultichainToken(currencyIn) || !currencyIn.hyperlaneAddress`